### ISSUE TITLE
All spaces after/before `%header.set(subject," text ")%` are trimmed away

### ISSUE THREAD

**Issue by PsLanger** [RealRaven2000/SmartTemplates#292]
Hi,
i've tried to add a space after the header subject text, but this doesn't work, even if i try to add two space.

In Detail:
I've made a template for new E-Mails with: %header.set(subject,"[Example] ")% but the space after ] doesn't appear in the header, even with two spaces or plain text without brackets.
Space seens to work only if there are character before an after it, at the beginning it also doesn't work.

Is this a bug?

Regards

**Comment by RealRaven2000** [comment]
Yeah it's possible, you can export the whole settings like this:

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/68265dc3-e771-4001-b320-943d3a8678ce)

this will export the templates for Write / Reply / Forward and any of the checkboxes on these pages.

- then I can test it directly here. I think generally `%header.set()%` should be "high-fidelity" - it should accept whatever you hand to it without change. Generally there may be some issues with spaces in HTML (encoding) but I don't think it applies to subject lines. I will test it with one of my templates as well next...

**Comment by RealRaven2000** [comment]
Yeah it is reproducible here, no need to send your template :

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/2f1ab6b6-8699-4a60-bc32-6ba1bafab75b)

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/d59f0e21-06a1-4695-89d1-541382931d8d)

**Comment by RealRaven2000** [comment]
Looks like there is a (likely unnecessary) call to `trim()` in my `modifyHeader` function:

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/b118e013-67f3-47ea-9257-2496e0a1570b)

The other purpose of the line is to replace potential whitespace / line breaks, I think this was designed to highlight an arbitrary portion from a text and copy it then paste it in the subject line (using the clipboard keyword), for example:

```
%header.append(subject,clipboard)%
```

by the way you can also concatenate multiple arguments, like so:

```
%header.set(subject,"[honeybadgers] ","Hi ", clipboard," here is your latest news!")%
```

Of course calling `trim() `may still have it's unwanted side effects here, so I will remove it for now, I don't think it will create any huge side effects:

[smartTemplate-fx-4.5pre74.zip](https://github.com/RealRaven2000/SmartTemplates/files/15273976/smartTemplate-fx-4.5pre74.zip)

------

To install the test version above, download the zip file and drag the it into Thunderbird Add-ons Manager (do not extract contents, it won't install like that)

**Comment by PsLanger** [comment]
Seems to be working, nice.
Already bought pro yesterday for great service. ;)

My second idea was an variable to force space like &nbsp; (what it already has done betreen characters/words if you don't format in html).
Could be a solution if the trim should be necessary anyway.

**Comment by RealRaven2000** [comment]
> Seems to be working, nice. Already bought pro yesterday for great service. ;)
> 
> My second idea was an variable to force space like   (what it already has done betreen characters/words if you don't format in html). Could be a solution if the trim should be necessary anyway.

yes, that may be a solution, as it is what the name processing - such as `%recipient(firstname,lastname,mail,work.department,work.organization)%` - already does. It simply concatenates the elements it can find in the addressbook and delimits them with a space. By the way another pitfall with spaces, my example in the previous comment had a mistake; SmartTemplates generally doesn't like the space after a comma that delimits parameters, and can lead to wrong results.

**Comment by RealRaven2000** [comment]
Fixed in 4.5 released on 14/05/2024
